File: Implementation/BasicMath/java/harmonicProgression.java
import java.util.*;

public class harmonicProgression
{
    static boolean isHP(double arr[])
    {
        if (arr.length == 1)
            return true;

        ArrayList<Integer> rec = new ArrayList<Integer>();
        for (int i = 0; i < arr.length; ++i)
            rec.add((int)(1 / arr[i]));
        Collections.sort(rec);
        int d = (int)rec.get(1) - (int)rec.get(0);
        for (int i = 2; i < arr.length; ++i)
            if (rec.get(i) - rec.get(i - 1) != d)
                return false;

        return true;
    }

    public static void main(String[] args)
    {
        double arr[] = {1 / 5, 1 / 10, 1 / 15, 1 / 20, 1 / 25};
        
        if (isHP(arr))
            System.out.println("Yes");
        else
            System.out.println("No");
    }
}

File: Implementation/BasicMath/cpp/harmonicProgression.cpp
#include <bits/stdc++.h>

bool isHP(vector<double> &arr)
{
    if(arr.size() == 1) return false;

    //Get the reciprocals
    vector<int> rec;
    for (int i = 0; i < arr.size(); ++i)
        rec.push_back((1 / (arr[i])));
    sort(rec.begin(), rec.end());
    int d = rec[1] - rec[0];

    for (int i = 2; i < arr.size(); ++i)
        if (rec[i] - rec[i - 1] != d) return false;

    return true;
}

using namespace std;

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    vector<double> arr = {1 / 5, 1 / 10, 1 / 15, 1 / 20, 1 / 25};
    
    (isHP(arr)) ? cout << "Yes" << "\n" : cout << "No" << "\n"; 

    return 0;
}
